app/assets/javascripts/upload-progress.js

```diff
@@ -0,0 +1,151 @@
+        document.attachEvent('onreadystatechange', function() {
+
+            if (document.readyState !== 'loading') fn();
+        });
+    }
+}
+
+function submitError(error, data) {
+
+    const payload = {
+        code: error,
+        values: [],
+        errorDetail: data
+    };
+    const xhr = new XMLHttpRequest();
+    xhr.onload = function() {
+
+        window.location = dac6UploadFormRedirect.val();
+    };
+    xhr.open('POST', dac6UploadReportError.val());
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.send(JSON.stringify(payload));
+}
+
+function fileUpload(form, url) {
+
+    const xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+
+        if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+
+            refreshStatusPage(url);
+        }
+    };
+    xhr.open('POST', form.action);
+    xhr.withCredentials = true;
+    xhr.send(new FormData(form));
+}
+
+function startSpinner(message) {
+
+    const fileElement = document.getElementById("file-upload");
+    const submitButtonElement = document.getElementById("submit");
+    const spinnerHtml = "<div id=\"processing\" aria-live=\"polite\" class=\"govuk-!-margin-bottom-5\">" +
+        "<h2 class=\"govuk-heading-m\">"+ message+ "</h2>" + // TODO i18n
+        "<div><div class=\"ccms-loader\"></div></div></div>";
+    fileElement.insertAdjacentHTML('beforebegin', spinnerHtml);
+    fileElement.setAttribute('disabled', 'disabled');
+    submitButtonElement.classList.add('govuk-button--disabled')
+}
+
+function stopSpinner(message) {
+
+    console.debug("Reached final status, removing refresh");
+    clearInterval(window.refreshIntervalId);
+    const spinnerElement = document.getElementById("processing");
+    spinnerElement.innerHTML = message;
+    console.debug("interval cleared");
+}
+
+function refreshStatusPage(url) {
+
+    const xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+        if (this.readyState === XMLHttpRequest.DONE) {
+
+            const dac6UploadStatusElement = document.getElementById("dac6UploadStatus");
+            if (this.status === 200) {
+
+                const data = JSON.parse(this.response);
+                if (dac6UploadStatusElement.value !== data._type) {
+
+                    console.info("status changed, updating page", data._type);
+                    dac6UploadStatusElement.value = data._type;
+
+                    console.info("page updated");
+                    if (data._type === "UploadedSuccessfully") {
+
+                        stopSpinner("<p>"+ data.name+ msg2.value+ "</p>");
+                    }
+                    else if (data._type === "Quarantined") {
+
+                        stopSpinner("<p>"+ data.name+ msg3.value+ "</p>"); // TODO improve message
+                    }
+                    else if (data._type === "Failed") {
+
+                        stopSpinner("<p>"+ data.name+ msg4.value+ "</p>");
+                    }
```

Comment:
are these not error pages that we're going to send people to? It also reduces the need to sanatize the input